### PR TITLE
Add py.typed export per PEP 561

### PR DESCRIPTION
**Pull request by maroux**
## Pull Request Checklist

Thank you for taking the time to improve Arrow! Before submitting your pull request, please check all *appropriate* boxes:

- [ ] 🧪  Added **tests** for changed code.
- [x] 🛠️  All tests **pass** when run locally (run `tox` or `make test` to find out!).
- [x] 🧹  All linting checks **pass** when run locally (run `tox -e lint` or `make lint` to find out!).
- [ ] 📚  Updated **documentation** for changed code.
- [x] ⏩  Code is **up-to-date** with the `master` branch.

If you have *any* questions about your code changes or any of the points above, please submit your questions along with the pull request and we will try our best to help!

## Description of Changes

Add `py.typed` file and export type annotations per [PEP 561](https://www.python.org/dev/peps/pep-0561/) so that type testing tools like mypy can actually verify types when using arrow.

No new test added for this change and I don't think there's any documentation necessary but let me know if something needs to be added.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -19,6 +19,7 @@
     author_email="crsmithdev@gmail.com",
     license="Apache 2.0",
     packages=["arrow"],
+    package_data={"arrow": ["py.typed"]},
```
Comment (jadchaar) [inline setup.py:22]: Might be better to specify this in `MANIFEST.in`: https://github.com/arrow-py/arrow/blob/master/MANIFEST.in

Comment (jadchaar) [reply setup.py:22]: Actually looking at PEP 561, this seems like the way to specify it. Disregard.

**Review by jadchaar — APPROVED** [review]
LGTM, but I had a quick question @maroux. If PEP 561 is Python 3.7+ compatible, will mypy not detect the typings in Python 3.6?

**Comment by maroux** [comment]
@jadchaar possible? At least for mypy, it looks like it supports `py.typed` even for py3.6:
```
○ → cat a.py
import arrow


if __name__ == "__main__":
    print(arrow.get(object()))


○ → python --version
Python 3.6.11


○ → mypy a.py
a.py:5: error: No overload variant of "get" matches argument type "object"
a.py:5: note: Possible overload variant:
a.py:5: note:     def get(Union[Arrow, datetime, date, struct_time, tzinfo, int, float, str, Tuple[int, int, int]], *, locale: str = ..., tzinfo: Union[tzinfo, str, None] = ..., normalize_whitespace: bool = ...) -> Arrow
a.py:5: note:     <3 more non-matching overloads not shown>
Found 1 error in 1 file (checked 1 source file)

```

I think for PEPs that don't add features to python core, the `python-version` isn't a requirement, but I could be wrong.

**Comment by jadchaar** [comment]
@isac322 you might be more familiar with PEP 561 for type annotations. Think this is good to merge?

**Comment by isac322** [comment]
@jadchaar Yep I forgot to add this 😅

**Comment by jadchaar** [comment]
> @jadchaar Yep I forgot to add this 😅

All good, just wanted to make sure!

**Pull request merged**